worldedit-core/src/main/java/com/sk89q/worldedit/util/SideEffectSet.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.util;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class SideEffectSet {
+    private static final SideEffectSet DEFAULT = new SideEffectSet(
+            Arrays.stream(SideEffect.values()).collect(Collectors.toMap(Function.identity(), SideEffect::getDefaultValue))
+    );
+    private static final SideEffectSet NONE = new SideEffectSet();
+
+    private final Map<SideEffect, SideEffect.State> sideEffects;
+    private final Set<SideEffect> appliedSideEffects;
+    private boolean appliesAny;
+
+    private SideEffectSet() {
+        this(ImmutableMap.of());
+    }
+
+    public SideEffectSet(Map<SideEffect, SideEffect.State> sideEffects) {
+        this.sideEffects = Maps.immutableEnumMap(sideEffects);
+
+        appliesAny = sideEffects.values().stream().anyMatch(state -> state != SideEffect.State.OFF);
```

Comment:
can't this line just be moved below the next and read `appliesAny = !appliedSideEffects.isEmpty()`?